Tasks/Doroshevich/Task6/Task6_permutation.py

```diff
@@ -0,0 +1,19 @@
+def permutation(s):
+    
+    if len(s) == 1:
+        return [s]
+
+    l = []
+
+    for i in s:
+        for j in permutation(s.replace(i, '', 1)):
+            if i+j not in l:
+                l.append(i + j) 
+
+    return l
+
+print(permutation('abba'))
```

Comment:
Please note, that despite being very convenient in terms of implementation the recursive function is not the most optimal way to find permutations. However, it is generally considered to be an acceptable solution for small to moderate sized strings.